### DISCUSSION TITLE
SemaphoreUI on Podman - folder permissions issue

### DISCUSSION THREAD

**Discussion by Pshemas** [semaphoreui/semaphore#2055] [Q&A]
I'm trying to get a nice setup of Caddy running in a container managed with Podman. I can get it running but I have some issues with folder permissions.

With default setup if I create `semaphore-data` folder and map it to `/etc/semaphore` , add config.json etc the process in a container can't use it.
If I don't create those in advance those are getting added upon container creation. But they belong to user `232070` ... which is also problematic if I want to add my own config file / edit existing one.
If I create those advance and in `compose.yaml` I specify UID, GUID to be the same as user starting the containers Semaphore fails as it can't write to `/tmp/semaphore` , `/home/semaphore` (and maybe some other, but I've just started testing).

For the time being I've mapped those folders in the volumes section. Is there a better / cleaner approach?

**Comment by tboerger** [comment]
This is not an issue really related to Semaphore but generally to podman.

For mounting volumes your got to add :Z to get it writable.

**Reply by Pshemas** [reply]
I already do that - that's how I can write to them at all. The problem is that when the SemaphoreUI images start they create folders for user 232070. But at the same time if I specify that I wish the UID to be the same as the user that runs the container I can't write to /tmp , /home/semaphore .

![obraz](https://github.com/semaphoreui/semaphore/assets/6144330/16911018-94cc-4271-b7a2-a8684d9dda50)


 And I wonder whether there's some clean solution to this.

**Reply by tboerger** [reply]
If you customize (mount) the working directory use a different path as /tmp got some special behavior.
